doc/develop/west/build-flash-debug.rst

```diff
@@ -311,6 +337,63 @@ For example, to build with 4 cores::
 
 The ``-o`` option is described further in the previous section.
 
+.. _west-domains-file:
+
+Domains file
+------------
+
+A domains file specifies multiple domains which can be built, flashed, and
+debugged individually using the ``--domain <domain>`` argument.
+
+The domains file is named ``domains.yaml`` and must be located in the top-level
+build folder of a multi-domain build.
+
+The ``domains.yaml`` contains domains along with their build directory.
+A ``default`` entry exists in the file.
+Commands which only handle a single domain, such as ``west debug``, use this
+entry to select the domain to use if no domain is specified.
+
+The following is a sample ``domains.yaml`` file:
+
+.. code-block:: yaml
+
+  default: <domain-n>
+  build_dir: <toplevel-build-dir>
+  domains:
+    - name: <domain-name-1>:
+      build_dir: <build-dir-domain-1>
+    - name: <domain-name-2>:
+      build_dir: <build-dir-domain-2>
+
+For example, the :ref:`hello_world` sample built with `MCUboot`_
+support will look as follows:
+
+.. code-block:: yaml
+
+  default: hello_world
+  build_dir: <toplevel-build-dir>
+  domains:
+    - name: mcuboot
+      build_dir: <toplevel-build-dir>/mcuboot
+    - name: hello_world
+      build_dir: <toplevel-build-dir>/hello_world
+
+
+Sysbuild
```

Comment:
Same issue here; this is under "Examples" too. An additional issue is that the content in this section is not just about west build, but is about the other west extensions -- like `flash` and `debug` -- documented on this page. The structure here really doesn't make sense.

Reply:
Moved and cleaned up.